rust/agama-lib/src/questions.rs

```diff
@@ -0,0 +1,71 @@
+use log::info;
+use std::collections::HashMap;
+
+/// module holdings data model for agama questions
+
+/// Basic generic question that fits question without special needs
+#[derive(Clone, Debug)]
+pub struct GenericQuestion {
+    /// numeric id used to indetify question on dbus
+    pub id: u32,
+    /// class of questions. Similar kinds of questions share same class.
+    /// It is dot separated list of elements. Examples are
+    /// `storage.luks.actication` or `software.repositories.unknown_gpg`
+    pub class: String,
+    /// Textual representation of question. Expected to be read by people
+    pub text: String,
+    /// possible answers for question
+    pub options: Vec<String>,
+    /// default answer. Can be used as hint or preselection and it is used as answer for unattended questions.
+    pub default_option: String,
+    /// additional data to help identify questions. Useful for automatic answers. It is question specific.
+    pub data: HashMap<String, String>,
+    /// Confirmed answer. If empty then not answered yet.
+    pub answer: String,
+}
+
+impl GenericQuestion {
+    pub fn new(
+        id: u32,
+        class: String,
+        text: String,
+        options: Vec<String>,
+        default_option: String,
+        data: HashMap<String, String>,
+    ) -> Self {
+        info!("Creating new generic question with text {}", text);
+        Self {
+            id,
+            class,
+            text,
+            options,
+            default_option,
+            data,
+            answer: String::from(""),
+        }
+    }
+
+    pub fn object_path(&self) -> String {
```

Comment:
Questions are coupled with D-Bus, right? I wonder whether this struct should live in `agama-lib` or directly in `agama-dbus-server`. After all, any other component should use the D-Bus API to deal with questions.

Reply:
@imobachgs reason why it is is in `agama-lib` and not in `agama-dbus-server` is that object path is useful also for client side. So my general rule is if it is useful for server and! client that have it in lib, so both CLI and dbus server can use it. If it is useful only for one of it, then it can live there.

Reply:
OK, it is fine. I tend to move the stuff to `agama-lib` only when I use it from other parts, but if you find it useful, it is OK.